src/main/kotlin/io/github/gradlenexus/publishplugin/NexusPublishPlugin.kt

```diff
@@ -261,3 +317,8 @@ class NexusPublishPlugin : Plugin<Project> {
         }
     }
 }
+
+inline fun <reified T : Any> Project.theExtension(): T =
+    typeOf<T>().let {
+        this.extensions.findByType(it) ?: throw IllegalStateException("The plugion cannot be applied without the publishing plugin")
```

Comment:
```suggestion
        this.extensions.findByType(it) ?: throw IllegalStateException("The plugin cannot be applied without the publishing plugin")
```

Also: the error message looks strange considering the extension fun's intent.